scripts/rpc/getSaleHistory.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * An example of how to query the Foundation sale history of an NFT using RPC (e.g. Alchemy) requests.
+ * Run with `yarn run-script scripts/rpc/getSaleHistory.ts --network mainnet`
+ */
+
+import { ethers } from "hardhat";
+import { FNDNFTMarket__factory } from "../../typechain-types";
+import { ReserveAuctionFinalizedEvent } from "../../typechain-types/FNDNFTMarket";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const addresses = require("@f8n/fnd-protocol/addresses.js");
+
+// The example NFT to query for, this should be updated with the NFT of interest
+const collection = "0x3B3ee1931Dc30C1957379FAc9aba94D1C48a5405";
+const tokenId = 105912;
+
+async function main(): Promise<void> {
+  // Connect to the Foundation market contract
+  const market = FNDNFTMarket__factory.connect(addresses.prod[1].nftMarket, ethers.provider);
```

Comment:
to be less fragile would it be possible to get the chainid using ethers?

Reply:
We could of course. But the contract is also mainnet specific. Just wanted to keep the script simple where we can.